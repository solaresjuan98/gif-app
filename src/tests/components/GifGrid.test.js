import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from "enzyme";
import { GifGrid } from "../../compontents/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('tests on <GifGrid />', () => {

    const category = 'Nintendo';

    test('should to show component correctly', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();


    });

    test('should to show items when images are charged useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/anything/a.jpg',
            title: 'Random thing'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)

    });


})
