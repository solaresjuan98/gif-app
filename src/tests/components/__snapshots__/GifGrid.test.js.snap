// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <GifGrid /> should to show component correctly 1`] = `
<Fragment>
  <h3>
    Nintendo
  </h3>
  <p
    className="animate__animated animate__flash "
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`tests on <GifGrid /> should to show items when images are charged useFetchGifs 1`] = `
<Fragment>
  <h3>
    Nintendo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Random thing"
      url="https://localhost/anything/a.jpg"
    />
  </div>
</Fragment>
`;
